lib/plugins/system/404/404.php

```diff
@@ -27,14 +27,21 @@ public static function handleError(JException $error)
 
         if ($app->get('sef') && (int) $error->getCode() == 404)
         {
+            $menu        = $app->getMenu()->getActive();
+            $menu_prefix = JRoute::_($menu->link, false);
+            $subpath     = preg_replace('#'.preg_quote(JURI::base(true)).'#', '', $menu_prefix, 1);
```

Comment:
having the replace=1 prefix is good but I'd still like to make sure we only replace the path from the beginning. Can you make the regexp start with a `^` please? Also if `JURI::base(true)` returns `/` this doesn't have to run right? We could add that for cleaner code as well.

Reply:
@ercanozkaya the reason i implemented that way is to replace all `/id-` in one call and to lessen it to just one redirection. I can also apply the regex from PlgSystemTextman to have less code to maintain, what do you suggest? :)

Reply:
@allanpilarca Let's go with the code in Textman if possible. The two redirections look like overhead but they do support an edge case. For example `/movies/12-angry-men/3423-cast.html` would be redirected to `/movies/angry-men/cast.html`without it which breaks the link.

Reply:
@ercanozkaya oki, i applied the regex code from TEXTman as you suggested.